src/java/main/org/apache/zookeeper/Login.java

```diff
@@ -198,8 +199,8 @@ public void run() {
                     }
                     if (isUsingTicketCache) {
                         String cmd = "/usr/bin/kinit";
-                        if (System.getProperty("zookeeper.kinit") != null) {
-                            cmd = System.getProperty("zookeeper.kinit");
+                        if (config.getProperty(ZKConfig.KINIT_COMMAND) != null) {
```

Comment:
Handeled it but slightly differently
`String cmd = config.getProperty(ZKConfig.KINIT_COMMAND, KINIT_COMMAND_DEFAULT);`